src/components/FileManager/FileManager.tsx

```diff
@@ -60,7 +61,10 @@ export const FileManager: React.FC = () => {
     inspectorOpenTabs: s.inspectorOpenTabs,
   }))
 
-  const { figures, supplements, attachments, others } = groupFiles(doc, files)
+  const { figures, supplements, attachments, linkedFiles, others } = groupFiles(
```

Comment:
this function iterates the entire document. Can you please add memo here to avoid recalculating it if doc or files didnt change?